src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/RSACryptoServiceProvider.Unix.cs

```diff
@@ -0,0 +1,241 @@
+            else
+            {
+                throw PaddingModeNotSupported();
+            }
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                _impl.Dispose();
+                base.Dispose(disposing);
+            }
+        }
+
+        public byte[] Encrypt(byte[] rgb, bool fOAEP)
+        {
+            if (rgb == null)
+            {
+                throw new ArgumentNullException(nameof(rgb));
+            }
+
+            return _impl.Encrypt(rgb, fOAEP ? RSAEncryptionPadding.OaepSHA1 : RSAEncryptionPadding.Pkcs1);
+        }
+
+        public override byte[] Encrypt(byte[] data, RSAEncryptionPadding padding)
+        {
+            if (data == null)
+                throw new ArgumentNullException(nameof(data));
+            if (padding == null)
+                throw new ArgumentNullException(nameof(padding));
+
+            if (padding == RSAEncryptionPadding.Pkcs1)
+            {
+                return Encrypt(data, fOAEP: false);
+            }
+            else if (padding == RSAEncryptionPadding.OaepSHA1)
+            {
+                return Encrypt(data, fOAEP: true);
+            }
+            else
+            {
+                throw PaddingModeNotSupported();
+            }
+        }
+
+        public byte[] ExportCspBlob(bool includePrivateParameters)
+        {
+            RSAParameters parameters = ExportParameters(includePrivateParameters);
+            return CapiHelper.ToKeyBlob(parameters, CapiHelper.CALG_RSA_KEYX);
+        }
+
+        public override RSAParameters ExportParameters(bool includePrivateParameters) =>
+            _impl.ExportParameters(includePrivateParameters);
+
+        protected override byte[] HashData(byte[] data, int offset, int count, HashAlgorithmName hashAlgorithm) =>
+            AsymmetricAlgorithmHelpers.HashData(data, offset, count, hashAlgorithm);
+
+        protected override byte[] HashData(Stream data, HashAlgorithmName hashAlgorithm) =>
+            AsymmetricAlgorithmHelpers.HashData(data, hashAlgorithm);
+
+        public override void FromXmlString(string xmlString) => _impl.FromXmlString(xmlString);
+
+        public void ImportCspBlob(byte[] keyBlob)
+        {
+            bool includePrivateParameters = CapiHelper.GetKeyBlobType(keyBlob) != CapiHelper.PUBLICKEYBLOB;
+            RSAParameters parameters = CapiHelper.ToRSAParameters(keyBlob, includePrivateParameters);
+            ImportParameters(parameters);
+        }
+
+        public override void ImportParameters(RSAParameters parameters)
+        {
+            // Although _impl supports larger Exponent, limit here for compat.
+            if (parameters.Exponent == null || parameters.Exponent.Length > 4)
+                throw new CryptographicException(SR.Argument_InvalidValue);
+
+            _impl.ImportParameters(parameters);
+
+            _publicOnly = (parameters.P == null || parameters.P.Length == 0);
```

Comment:
`D` is what normally gets checked.  Also, is D (or P).Length == 0 a valid case?

And a comment about _impl.ImportParameters having already verified that the private key parameters are all-or-none wouldn't go amiss.